### ISSUE TITLE
Event-Driven SD: pool members deleted when monitor changed

### ISSUE THREAD

**Issue by chen23** [F5Networks/f5-appsvcs-extension#150]
##### ISSUE TYPE

 - Bug Report
 

##### AS3 BUILD/ VERSION

```
{"version":"3.14.0","release":"4","schemaCurrent":"3.14.0","schemaMinimum":"3.0.0"}
{
  "release": "4",
  "schemaCurrent": "3.14.0",
  "schemaMinimum": "3.0.0",
  "version": "3.14.0"
}
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.1.2
  Build       0.0.4
  Edition     Point Release 2
  Date        Thu Oct 11 15:32:21 PDT 2018
```

##### SUMMARY

when you make a change to the pool monitor any previous pool members that were created by event-driven service discovery are deleted.

##### STEPS TO REPRODUCE

Declaration 1

```
{
    "schemaVersion": "3.9.0",
    "class": "ADC",
    "Tenant": {
        "class": "Tenant",
        "Application": {
            "class": "Application",
            "template": "generic",
            "pool": {
                "class": "Pool",
                "monitors": ["http"],
                "members": [
                    {
                        "addressDiscovery": "event",
                        "servicePort": 80
                    }
                ]
            }
        }
    }
}
```
Update to SD task
```
[
    {
        "id": "newNode1",
        "ip": "192.0.2.3"
    },
    {
        "id": "NewNode2",
        "ip": "192.0.2.4"
    }
]
```
TMSH output
```
ltm pool Application/pool {
    members {
        NewNode2:80 {
            address 192.0.2.4
            session monitor-enabled
            state down
        }
        newNode1:80 {
            address 192.0.2.3
            session monitor-enabled
            state down
        }
    }
    min-active-members 1
    monitor min 1 of { /Common/http }
    partition Tenant
}
```
Declaration 2
```
{
    "schemaVersion": "3.9.0",
    "class": "ADC",
    "Tenant": {
        "class": "Tenant",
        "Application": {
            "class": "Application",
            "template": "generic",
            "pool": {
                "class": "Pool",
                "monitors": ["tcp"],
                "members": [
                    {
                        "addressDiscovery": "event",
                        "servicePort": 80
                    }
                ]
            }
        }
    }
}
```
TMSH output
```
ltm pool Application/pool {
    min-active-members 1
    monitor min 1 of { /Common/tcp }
    partition Tenant
}
```

##### EXPECTED RESULTS

preserve existing pool members

##### ACTUAL RESULTS

existing pool members are deleted

**Comment by dstokesf5** [comment]
This has already been fixed, and the fix should be available in the 3.15 release.

**Comment by dstokesf5** [comment]
After further testing, this does appear to be different then the bug already fixed for 3.15. I have added this to our internal product backlog as AT-756.

**Comment by chen23** [comment]
@dstokesf5 looks like this got fixed at some point?  tested with v3.21
```
ltm pool Application/pool {
    members {
        NewNode2:80 {
            address 192.0.2.4
            session monitor-enabled
            state down
        }
        newNode1:80 {
            address 192.0.2.3
            session monitor-enabled
            state down
        }
    }
    min-active-members 1
    monitor min 1 of { /Common/tcp }
    partition Tenant
}
```

**Comment by dstokesf5** [comment]
Looks like AUTOTOOL-756 is marked as resolved in 3.16. Thank you for confirming that the issue was resolved.
